### PR TITLE
README fixes, includes specifics for PyCharm configuration

### LINKED ISSUE #171: Distinction from developer setup and personal setup / Fixes to README
#### Fixing README to distinguish setups. 
* Specify bash interpreter
* Distinction between steps 2-4 for the two setups.
* Some specifics for PyCharm users.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/AZUL-127)
┆Project Name: azul
┆Issue Number: AZUL-127

### PR DESCRIPTION
**Pull request by achave11**
Resolves #171

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,21 +43,53 @@ generic with minimal need for project-specific behavior.
 The remaining infrastructure is managed internally with TerraForm.
 
 ### 1.3. Project configuration
+Getting started with out attempting to make major contributions, a set of the test suite passes without AWS credentials.
```
Comment (hannes-ucsc) [inline README.md]: Getting started without attempting to make contributions does not require AWS credentials. A subset of the test suite passes without configured AWS credentials. To validate your setup, we'll be running one of those tests at the end.

File: README.md [hunk 2]
```diff
@@ -41,21 +43,53 @@ generic with minimal need for project-specific behavior.
...
    
    ```
-   virtualenv .venv
+   cd azul
+   virtualenv -p python3 .venv
    source .venv/bin/activate
    ```
 
-2) Choose a name for your personal deployment. The name should be a short
+1) Setup configuration for dev deployment:
```
Comment (hannes-ucsc) [inline README.md]: Insert empty line.

File: README.md [hunk 3]
```diff
@@ -41,21 +43,53 @@ generic with minimal need for project-specific behavior.
...
+1) Setup configuration for dev deployment: 
+   ```
+   cd deployments
+   ln -snf dev .active
+   cd ..
+   ```
+
+2) Load the environment:
+
+    ```
```
Comment (hannes-ucsc) [inline README.md]: Three spaces of indent, not four.

File: README.md [hunk 4]
```diff
@@ -41,21 +43,53 @@ generic with minimal need for project-specific behavior.
...
+    source environment
+    ```
+   
+    Examine the output.
+
+3) Install the development prerequisites
+
+    ```
```
Comment (hannes-ucsc) [inline README.md]: Three spaces of indent, not four.

File: README.md (hunk 1 above)
Comment (hannes-ucsc) [reply README.md]: Wrap paragraphs at exactly 80 columns.

File: README.md [hunk 5]
```diff
@@ -41,21 +43,53 @@ generic with minimal need for project-specific behavior.
...
+    pip install -r requirements.dev.txt
+    ```
+
+4) Run `make`. It should say `Looking good!` If one of the sanity checks fails,
+   address the complaint and repeat. The various sanity checks are defined in
+   `common.mk`.
+
+---
```
Comment (hannes-ucsc) [inline README.md]: remove

File: README.md [hunk 6]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
    cd ..
    ```  
 
-4) Edit `deployments/.active/environment` and
+3) Edit `deployments/.active/environment` and
    `deployments/.active/environment.local` according to the comments in there.
 
-5) Load the environment:
+___
```
Comment (hannes-ucsc) [inline README.md]: remove

File: README.md [hunk 7]
```diff
@@ -41,21 +43,53 @@ generic with minimal need for project-specific behavior.
...
+    pip install -r requirements.dev.txt
+    ```
+
+4) Run `make`. It should say `Looking good!` If one of the sanity checks fails,
+   address the complaint and repeat. The various sanity checks are defined in
+   `common.mk`.
+
+---
+
+#### 1.3.1 For personal deployment (AWS credentials available)
```
Comment (hannes-ucsc) [inline README.md]: Add: Creating a personal deployment of Azul allows you test changes on a live system in complete isolation from other users. If you intend to make contributions, this is preferred. You will need IAM user credentials to the AWS account you are deploying to.

File: README.md [hunk 8]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
...
 
-5) Load the environment:
+___
 
-   ```
-   source environment
-   ```
-   
-   Examine the output.
+### 1.4 PyCharm configuration specifics.
```
Comment (hannes-ucsc) [inline README.md]: > 1.4. Configuring the project in PyCharm

(dot after the 4 and no dot and the end of headings)

File: README.md [hunk 9]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
...
+Configuring the working environment in PyCharm can improve the IDEs assistance. This allows for the whole environment to be synchronize and eliminate issues pertaining to dependencies.
```
Comment (hannes-ucsc) [inline README.md]: > Running tests from PyCharm requires `environment` to be sourced. The easiest way to do this is to install `envhook.py`, a helper script that injects the environment variables from `environment` into the Python interpreter process started from the project's `virtualenv` in `.venv`:

File: README.md [hunk 10]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
...
-   Examine the output.
+### 1.4 PyCharm configuration specifics.
+Configuring the working environment in PyCharm can improve the IDEs assistance. This allows for the whole environment to be synchronize and eliminate issues pertaining to dependencies.   
 
-6) Install the development prerequisites
-   
-   ```
-   pip install -r requirements.dev.txt
-   ```
+- Connect python hook for `environment` in PyCharm
```
Comment (hannes-ucsc) [inline README.md]: Remove

File: README.md [hunk 11]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
...
+    ```bash
```
Comment (hannes-ucsc) [inline README.md]: The `bash` thing doesn't seem to be doing anything.

File: README.md [hunk 12]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
...
+    python scripts/envhook.py install
+    ```
 
-7) Run `make`. It should say `Looking good!` If one of the sanity checks fails,
-   address the complaint and repeat. The various sanity checks are defined in
-   `common.mk`.
+- Under `Settings -> Project interpreter` select the virtual environment created above.
```
Comment (hannes-ucsc) [inline README.md]: `Project` and `Interpreter` should be italic and there should be an emdash (wide dash) between them. Alt-Shift-Dash. Should look like this: —

File: README.md [hunk 13]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
...
+   * Under show all, select `.venv/bin/python` if not already selected.
 
+
+- Set `src` & `test` folder as Source Root.
+   * Right click the folder name and select `Mark Directory as` `->` `Source Root`
+
+- Confirm proper installation, run the following.
```
Comment (hannes-ucsc) [inline README.md]: This should be the last step in 1.3 not here.

File: README.md [hunk 14]
```diff
@@ -64,27 +98,31 @@ The remaining infrastructure is managed internally with TerraForm.
...
+```shell
```
Comment (hannes-ucsc) [inline README.md]: And again, the shell thing doesn't seem to be having any effect.

File: environment [hunk 15]
```diff
@@ -199,5 +199,5 @@ _select () {
     ( cd ${AZUL_HOME}/deployments && ls -l .active )
 }
 
-export PYTHONPATH="$AZUL_HOME/src"
+export PYTHONPATH="$AZUL_HOME/src:$AZUL_HOME/test"
```
Comment (hannes-ucsc) [inline environment]: Separate commit.

**Comment by hannes-ucsc** [comment]
Should be two commits. When you go through my comments, mark each one as done by replying to it with "Fixed" or "Done".

**Comment by achave11** [comment]
Working rebase from PyCharm.

File: README.md [hunk 16]
```diff
@@ -41,21 +43,65 @@ generic with minimal need for project-specific behavior.
...
+   `common.mk`.
+   
+6) Confirm proper installation, run the following:
+   
+   ```
+   python scripts/test/indexer/test_hca_indexer.py
+   ``` 
+
+#### 1.3.1 For personal deployment (AWS credentials available)
+Creating a personal deployment of Azul allows you test changes on a live system
```
Comment (hannes-ucsc) [inline README.md]: Empty line before this one.

File: README.md [hunk 17]
```diff
@@ -64,27 +110,28 @@ The remaining infrastructure is managed internally with TerraForm.
...
 
-5) Load the environment:
 
-   ```
-   source environment
-   ```
-   
-   Examine the output.
+### 1.4. PyCharm configuration specifics
+Running tests from PyCharm requires `environment` to be sourced. The easiest way
```
Comment (hannes-ucsc) [inline README.md]: Empty line before.

File: README.md [hunk 18]
```diff
@@ -41,21 +43,65 @@ generic with minimal need for project-specific behavior.
...
+
+   ```
+   pip install -r requirements.dev.txt
+   ```
+
+5) Run `make`. It should say `Looking good!` If one of the sanity checks fails,
+   address the complaint and repeat. The various sanity checks are defined in
+   `common.mk`.
+   
+6) Confirm proper installation, run the following:
```
Comment (hannes-ucsc) [inline README.md]: `proper configuration`

File: README.md [hunk 19]
```diff
@@ -41,21 +43,66 @@ generic with minimal need for project-specific behavior.
 The remaining infrastructure is managed internally with TerraForm.
 
 ### 1.3. Project configuration
+Getting started without attempting to make contributions does not require AWS
```
Comment (hannes-ucsc) [inline README.md]: Empty line

**Pull request merged**